yarn-project/canary/src/uniswap_trade_on_l1_from_l2.test.ts

```diff
@@ -1,388 +1,231 @@
-  }, 60_000);
-  it('should uniswap trade on L1 from L2 funds privately (swaps WETH -> DAI)', async () => {
-    logger('Running L1/L2 messaging test on HTTP interface.');
-
-    [ownerWallet] = await getSandboxAccountsWallets(pxe);
-    const accounts = await ownerWallet.getRegisteredAccounts();
-    ownerAddress = accounts[0].address;
-    const receiver = accounts[1].address;
-
-    const result = await deployAllContracts(ownerWallet, ownerAddress, publicClient, walletClient);
-    const {
-      daiL2Contract,
-      daiL2Bridge,
-      daiContract,
-      daiTokenPortalAddress,
-      wethL2Contract,
-      wethL2Bridge,
-      wethContract,
-      wethTokenPortal,
-      wethTokenPortalAddress,
-      uniswapL2Contract,
-      uniswapPortal,
-    } = result;
-
-    const ownerInitialBalance = await getL2PrivateBalanceOf(ownerAddress, wethL2Contract);
-    logger(`Owner's initial L2 WETH balance: ${ownerInitialBalance}`);
 
+    logger('Deploying DAI Portal, initializing and deploying l2 contract...');
+    daiCrossChainHarness = await CrossChainTestHarness.new(
+      pxe,
+      publicClient,
+      walletClient,
+      ownerWallet,
+      logger,
+      DAI_ADDRESS,
+    );
+
+    logger('Deploying WETH Portal, initializing and deploying l2 contract...');
+    wethCrossChainHarness = await CrossChainTestHarness.new(
+      pxe,
+      publicClient,
+      walletClient,
+      ownerWallet,
+      logger,
+      WETH9_ADDRESS,
+    );
+
+    logger('Deploy Uniswap portal on L1 and L2...');
+    uniswapPortalAddress = await deployL1Contract(walletClient, publicClient, UniswapPortalAbi, UniswapPortalBytecode);
+    uniswapPortal = getContract({
+      address: uniswapPortalAddress.toString(),
+      abi: UniswapPortalAbi,
+      walletClient,
+      publicClient,
+    });
+    // deploy l2 uniswap contract and attach to portal
+    uniswapL2Contract = await UniswapContract.deploy(ownerWallet)
+      .send({ portalContract: uniswapPortalAddress })
+      .deployed();
+
+    const registryAddress = (await pxe.getNodeInfo()).l1ContractAddresses.registryAddress;
+    await uniswapPortal.write.initialize([registryAddress.toString(), uniswapL2Contract.address.toString()], {} as any);
+  });
+
+  beforeEach(async () => {
     // Give me some WETH so I can deposit to L2 and do the swap...
     logger('Getting some weth');
-    await walletClient.sendTransaction({
-      to: WETH9_ADDRESS.toString(),
-      value: parseEther('1'),
-    });
+    await walletClient.sendTransaction({ to: WETH9_ADDRESS.toString(), value: parseEther('1') });
+  });
 
-    const wethL1BeforeBalance = await wethContract.read.balanceOf([ownerEthAddress.toString()]);
-    // 1. Approve weth to be bridged
-    await wethContract.write.approve([wethTokenPortalAddress.toString(), wethAmountToBridge], {} as any);
+  it('should uniswap trade on L1 from L2 funds privately (swaps WETH -> DAI)', async () => {
```

Comment:
everything from this line onwards is an exact copy-paste of `end-to-end/uniswap_trade_on_l1_from_l2.test.ts`!